frontend/src/pages/integrations/railway/create.tsx

```diff
@@ -20,15 +21,27 @@ import {
 import { useGetWorkspaceById } from "../../../hooks/api/workspace";
 import createIntegration from "../../api/integrations/createIntegration";
 
+interface RailwayCreateIntegrationFormValues {
+  selectedSourceEnvironment: string;
+  secretPath: string;
+  targetAppId: string;
+  targetEnvironmentId: string;
+  targetServiceId: string;
+}
+
 export default function RailwayCreateIntegrationPage() {
   const router = useRouter();
+  const { handleSubmit, setValue, getValues, control, watch } =
+    useForm<RailwayCreateIntegrationFormValues>({
+      defaultValues: {
+        selectedSourceEnvironment: "",
+        secretPath: "",
+        targetAppId: "",
+        targetEnvironmentId: "",
+        targetServiceId: ""
+      }
+    });
 
-  const [targetAppId, setTargetAppId] = useState("");
-  const [targetEnvironmentId, setTargetEnvironmentId] = useState("");
-  const [targetServiceId, setTargetServiceId] = useState("");
-
-  const [selectedSourceEnvironment, setSelectedSourceEnvironment] = useState("");
-  const [secretPath, setSecretPath] = useState("/");
   const [isLoading, setIsLoading] = useState(false);
```

Comment:
There is a state inside useForm`formState:{isSubmitting}` use that instead of manual isLoading